claims_p1.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2021-2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package psatoken
+
+import (
+	_ "crypto/sha256" // used hash algorithms need to be imported explicitly
+	"fmt"
+)
+
+// P1Claims are associated with profile "PSA_IOT_PROFILE_1"
+type P1Claims struct {
+	Profile                *string        `cbor:"-75000,keyasint,omitempty" json:"psa-profile"`
+	ClientID               *int32         `cbor:"-75001,keyasint" json:"psa-client-id"`
+	SecurityLifeCycle      *uint16        `cbor:"-75002,keyasint" json:"psa-security-lifecycle"`
+	ImplID                 *[]byte        `cbor:"-75003,keyasint" json:"psa-implementation-id"`
+	BootSeed               *[]byte        `cbor:"-75004,keyasint" json:"psa-boot-seed"`
+	CertificationReference *string        `cbor:"-75005,keyasint,omitempty" json:"psa-certification-reference,omitempty"`
+	SwComponents           *[]SwComponent `cbor:"-75006,keyasint,omitempty" json:"psa-software-components,omitempty"`
+	NoSwMeasurements       *uint          `cbor:"-75007,keyasint,omitempty" json:"psa-no-software-measurements,omitempty"`
+	Nonce                  *[]byte        `cbor:"-75008,keyasint" json:"psa-nonce"`
+	InstID                 *[]byte        `cbor:"-75009,keyasint" json:"psa-instance-id"`
+	VSI                    *string        `cbor:"-75010,keyasint,omitempty" json:"psa-verification-service-indicator,omitempty"`
+}
+
+func newP1Claims(includeProfile bool) (IClaims, error) {
```

Comment:
In my view the includeProfile boolean in the interface makes it ambigous.

As the API itself is named after P1, then instantiation of P1 Claim should by default set the Profile to P1.

Should we further allow the user to set the Profile explicitly with SetProfile() API then :
(Strictly not needed in my view but OK if you want that flexibility)..

1. It should be OK to let the Profile parameter with Value PSA_IOT_PROFILE_1 over-write itself, why an error in that case??
2. If anything other than PSA-IOT_PROFILE_1 is set, then flag an Error and return.

Reply:
This is a private method.  `NewClaims` will "do the right thing" and instantiate the p1 claims with the profile claim in.

Reply:
Apologies, I think, I should start review from Interface file: Not the implementation.

Thanks for the clarification!